editors/code/src/persistent_state.ts

```diff
@@ -21,12 +21,15 @@ export class PersistentState {
      * Release id of the *nightly* extension.
      * Used to check if we should update.
      */
-    get releaseId(): number | undefined {
+    get nightlyReleaseId(): number | undefined {
         return this.globalState.get("releaseId");
     }
-    async updateReleaseId(value: number) {
+    async updateNightlyReleaseId(value: number) {
         await this.globalState.update("releaseId", value);
     }
+    async removeNightlyReleaseId() {
+        await this.globalState.update("releaseId", undefined);
+    }
```

Comment:
```suggestion
    async updateNightlyReleaseId(value: number | undfined) {
        await this.globalState.update("releaseId", value);
    }
```

Reply:
This feels a bit more dangerous with TS type system, but sure, done.

Reply:
Yeah, my rule of thumb here is to not break symmetry between `get/set` , if that can be avoided. 

The additional type-safety of not being able to pass `undefined`a to `update` is balanced out by the additional complexity and moving parts in the API. I don't know what would prevent more bugs, so I am biasing towards the simplest thing that can work.